packages/renderer/src/lib/help/HelpActions.svelte

```diff
@@ -0,0 +1,53 @@
+<script lang="ts">
+import { DropdownMenu } from '@podman-desktop/ui-svelte';
+
+import { ActionKind, type ItemAction, Items } from './HelpItems';
+import HelpMenu from './HelpMenu.svelte';
+
+let showMenu = false;
+let outsideWindow: HTMLDivElement;
+window.events?.receive('toggle-help-menu', toggleMenu);
+
+function toggleMenu(): void {
+  showMenu = !showMenu;
+}
+
+function handleEscape({ key }: KeyboardEvent): void {
+  if (key === 'Escape') {
+    showMenu = false;
+  }
+}
+
+function onWindowClick(e: any): void {
+  showMenu = outsideWindow.contains(e.target);
+}
+
+function createOnClick(action?: ItemAction): () => void | undefined {
```

Comment:
question: wondering why the method is not async ?

Reply:
Do you mean we should have something like this, so svelte would handle exceptions for us?
```
function createOnClick(action?: ItemAction): () => Promise<unknown> {
  return async () => {
    toggleMenu();
    if (action?.kind === ActionKind.LINK) {
      return window.openExternal(action.parameter);
    } else if (action?.kind === ActionKind.COMMAND) {
      return window.executeCommand(action.parameter);
    }
  };
}
```

Reply:
it's not possible to have directly the method being async and use await when calling the window methods ?



```
function async createOnClick(): Promise<unknown> {
  toggleMenu();
  if () {
     await window....
```

Reply:
the main point is to use await when calling the window. methods, not having to catch all the calls.

and yes svelte will display the error anyway in the console if the call is failing

Reply:
Update function to async and called it as `onClick{()  => onClick()}`.
The same pattern seems to be used everywhere.